config/routes.rb

```diff
@@ -85,5 +85,6 @@
   match '/', via: [:get], to: redirect("/api/v1")
 
   get 'pub/:token/:expires/:use/:class/:id/:version/:name.:extension' => 'public_assets#show', as: :public_asset, constraints: {name: /[^\/]+/}
+  match 'pub/*any', via: [:options], to: 'public_assets#options'
```

Comment:
I could be convinced of that.  Would have to duplicate (or more likely abstract) some of that shared HEAD/GET logic in the current `show` method.